pwa/src/components/endpoints/endpointForm.tsx

```diff
@@ -5,140 +5,116 @@ import {
   Alert,
   SelectInputComponent,
   Card,
-  Accordion
 } from "@conductionnl/nl-design-system/lib";
-import {isLoggedIn} from "../../services/auth";
 import {Link} from "gatsby";
 import {navigate} from "gatsby-link";
 import {
   checkValues,
   removeEmptyObjectValues,
 } from "../utility/inputHandler";
 import FlashMessage from 'react-flash-message';
-import LoadingOverlay from "../loadingOverlay/loadingOverlay";
+import APIService from "../../apiService/apiService";
+import APIContext from "../../apiService/apiContext";
+import LoadingOverlay from '../loadingOverlay/loadingOverlay'
 
 interface EndpointFormProps {
   id: string,
 }
 
 export const EndpointForm: React.FC<EndpointFormProps> = ({id}) => {
-
-  const [context, setContext] = React.useState(null);
+  const [alert, setAlert] = React.useState<Record<string, string>>(null);
   const [endpoint, setEndpoint] = React.useState<any>(null);
-  const [applications, setApplications] = React.useState<any>(null);
+  const [applications, setApplication] = React.useState<any>(null);
   const [showSpinner, setShowSpinner] = React.useState<boolean>(false);
+  const API: APIService = React.useContext(APIContext)
+  const title: string = id ? "Edit Endpoint" : "Create Endpoint";
   const [loadingOverlay, setLoadingOverlay] = React.useState<boolean>(false);
-  const [alert, setAlert] = React.useState<any>(null);
-  const title: string = (id === "new") ? "Create Endpoint" : "Edit Endpoint"
 
   React.useEffect(() => {
-    if (typeof window !== "undefined" && context === null) {
-      setContext({
-        adminUrl: process.env.GATSBY_ADMIN_URL,
-      });
-    } else if (isLoggedIn()) {
-      if (id !== "new") {
-        getEndpoint();
-      }
-      getApplications();
-    }
-  }, [context]);
+    handleSetApplications()
+    id && handleSetEndpoints()
+  }, [API, id])
 
-  const getEndpoint = () => {
-    setShowSpinner(true);
-    fetch(`${context.adminUrl}/endpoints/${id}`, {
-      credentials: "include",
-      headers: {"Content-Type": "application/json", 'Authorization': 'Bearer ' + sessionStorage.getItem('jwt')},
-    })
-      .then((response) => response.json())
-      .then((data) => {
-        setShowSpinner(false);
-        setEndpoint(data);
-      })
-      .catch((error) => {
-        setShowSpinner(false);
-        console.error("Error:", error);
-        setAlert(null);
-        setAlert({type: 'danger', message: error.message});
-      });
 
+  const handleSetEndpoints = () => {
```

Comment:
You're setting one endpoint, not multiple. Therefore, this function should be singular: `handletSetEndpoint`.